addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1533,25 +1533,26 @@ export class Wysiwyg extends Component {
         const targetEl = this.odooEditor.document.getSelection();
         const closest = closestBlock(targetEl.anchorNode);
         const restoreSelection = preserveCursor(this.odooEditor.document);
-        const blockWidth = parseInt(getComputedStyle(closest).width.slice(0, -2)); // remove 'px'
-        // Calculate one/fouth portion of blockWidth, and then on the basis of our cursor position
-        // check in which quadrant it falls, give the popover its position accordingly
-        const portionWidth = (blockWidth / 4);
-        const cursorAt = getRangePosition(targetEl, this.odooEditor.document).left;
-        let position = 'end';
-        if (cursorAt < portionWidth * 2) {
-            position = 'start';
-        } else if (cursorAt <= portionWidth * 3) {
-            position = 'middle';
-        }
 
         this.popover.add(closest, EmojiPicker, {
                 onSelect: (str) => {
                     restoreSelection();
                     this.odooEditor.execCommand('insert', str);
                 },
+            }, {
+                onPositioned: (popover) => {
+                    restoreSelection();
+                    let options = {};
+                    if (this.options.getContextFromParentRect) {
+                        options['parentContextRect'] = this.options.getContextFromParentRect();
+                    }
```

Comment:
IMO this piece serves no purpose here, since we are not using it here,  also there is room for this code to be optimized.
i suggest this code will also do the job :
`                onPositioned: (popover) => {
                    restoreSelection();
                    const rangePosition = getRangePosition(popover, this.options.document, {});
                    popover.style.top = rangePosition.top + 'px';
                    popover.style.left = rangePosition.left + 'px';
                    const oInputBox = popover.getElementsByClassName('o_input')[0];
                    oInputBox.focus();
                },`
@dhba-odoo can you check it let me know.

Reply:
@shsa-odoo I actually require the code as it is. In the 'getDeepRange' method, there is an 'if' condition that is essential for handling the 'parentContextRect.' If this condition is removed, the code within 'getDeepRange' will not execute, leading to an incorrect positioning of the emoji picker in Email Marketing. Therefore, we cannot remove that code.

Reply:
I tried functionally by removing that code and it seems to work fine in editor. What am i missing ? is it missing something in email marketing ?